sdk/iot/hub/inc/az_iot_hub_client.h

```diff
@@ -403,7 +403,7 @@ AZ_NODISCARD az_result az_iot_hub_client_methods_get_subscribe_topic_filter(
  */
 typedef struct az_iot_hub_client_method_request
 {
-  az_span request_id; /**< The request id.
+  uint32_t request_id; /**< The request id.

```

Comment:
Make it `int32_t` instead?

Reply:
why waster one bit?

---
In reply to: [421090366](https://github.com/Azure/azure-sdk-for-c/pull/693#discussion_r421090366) [](ancestors = 421090366)

Reply:
Primarily for clearer overflow validation and I don't know if it is meaningful to have request ids in the 2^31 or 2^32 range. Are we intentionally relying on that extra bit for large ids?

Also, generally speaking, it isn't user friendly to mix signed/unsigned ints unless really needed.
https://youtu.be/_nqAsSlE_NI?t=2560

Reply:
Every number is valid as it is decided via user, why should sdk implementation care about it?

If user provides this value and we take it as unsigned number, how are you mixing it? Not sure what you mean by this.

Reply:
Overflow are not really an issue here. It's just a unique ID. It is not used in indexing into array for example

Reply:
@danewalton please confirm that the int on the hub side is 32bit unsigned.
@ahsonkhan in this case we need to use the same type as the service. Using a signed may lead to overflow and, in most implementations negative numbers which will result in failures caused by our SDK when encoding the responses.